facade/src/main/java/org/jboss/pnc/facade/providers/BuildConfigurationProviderImpl.java

```diff
@@ -347,16 +371,29 @@ public BuildConfigurationRevision createRevision(String id, BuildConfiguration b
                 bcsWithLatest);
     }
 
-    private BuildConfigurationWithLatestBuild populateBuildConfigWithLatestBuild(BuildConfiguration buildConfig) {
-        BuildPageInfo pageInfo = new BuildPageInfo(0, 10, null, null, true, false, buildConfig.getName());
-        Optional<Build> latestBuild = buildProvider.getBuildsForBuildConfiguration(pageInfo, buildConfig.getId())
-                .getContent()
+    private Optional<BuildTask> getLatestBuildTaskForBuildConfig(Integer configId) {
+        return buildCoordinator.getSubmittedBuildTasks()
                 .stream()
+                .filter(Objects::nonNull)
+                .filter(buildTask -> buildTask.getBuildConfigurationAudited().getId().equals(configId))
+                .max(Comparator.comparing(BuildTask::getSubmitTime));
+    }
+
+    private BuildConfigurationWithLatestBuild populateBuildConfigWithLatestBuild(
+            BuildConfiguration buildConfig,
+            List<BuildRecord> latestBuilds) {
+        Optional<BuildTask> latestBuildTask = getLatestBuildTaskForBuildConfig(
+                mapper.getIdMapper().toEntity(buildConfig.getId()));
+        Optional<BuildRecord> latestBuildRecord = latestBuilds.stream()
+                .filter(br -> br.getBuildConfigurationId().equals(mapper.getIdMapper().toEntity(buildConfig.getId())))
                 .findFirst();
+        Build latestBuild = latestBuildTask.map(buildMapper::fromBuildTask)
+                .orElse(latestBuildRecord.map(buildMapper::toDTO).orElse(null));
+
         return BuildConfigurationWithLatestBuild.builderWithLatestBuild()
                 .buildConfig(buildConfig)
-                .latestBuild(latestBuild.orElse(null))
-                .latestBuildUsername(latestBuild.map(build -> build.getUser().getUsername()).orElse(null))
+                .latestBuild(latestBuild)
+                .latestBuildUsername(latestBuild == null ? "" : latestBuild.getUser().getUsername())
```

Comment:
before you passed null when there was no build, now you pass empty string. Not sure which one is better.